packages/template-retail-react-app/app/utils/product-utils.test.js

```diff
@@ -9,69 +9,124 @@ import {
+            size: '065',
+            width: 'M'
+        }
+        const result = getDisplayVariationValues(variationAttributes, selectedValues)
 
-    expect(result).toEqual({
-        Colour: 'Taupe',
-        Size: '6.5',
-        Width: 'M'
+        expect(result).toEqual({
+            Colour: 'Taupe',
+            Size: '6.5',
+            Width: 'M'
+        })
     })
 })
 
 describe('getDisplayPrice', function () {
-    test('returns basePrice and discountPrice', () => {
-        const {basePrice, discountPrice} = getDisplayPrice(
-            mockedCustomerProductListsDetails.data[0]
-        )
+    test('returns listPrice and currentPrice for product that has only priceRanges', () => {
+        const data = {
+            name: 'product name',
+            price: 37.76,
+            priceRanges: [
+                {
+                    maxPrice: 40.76,
+                    minPrice: 30.76,
+                    pricebook: 'gbp-m-list-prices'
+                },
+                {
+                    maxPrice: 37.76,
+                    minPrice: 37.76,
+                    pricebook: 'gbp-m-sale-prices'
+                }
+            ]
+        }
+        const {listPrice, currentPrice} = getDisplayPrice(data)
+
+        expect(listPrice).toBe(40.76)
+        expect(currentPrice).toBe(37.76)
+    })
+
+    test('returns listPrice and currentPrice for product that has only tieredPrices', () => {
+        const data = {
+            name: 'product name',
+            price: 25.6,
+            tieredPrices: [
+                {
+                    price: 30.6,
+                    pricebook: 'gbp-m-list-prices',
+                    quantity: 1
+                },
+                {
+                    price: 25.6,
+                    pricebook: 'gbp-m-sale-prices',
+                    quantity: 1
+                }
+            ]
+        }
+        const {listPrice, currentPrice} = getDisplayPrice(data)
 
-        expect(basePrice).toBe(199.0)
-        expect(discountPrice).toBe(189.0)
+        expect(listPrice).toBe(30.6)
+        expect(currentPrice).toBe(25.6)
     })
 
-    test('returns null if there is not discount promotion', () => {
+    test('returns pick the closest tieredPrices for product currentPrice if quantity is more than 1', () => {
         const data = {
-            ...mockedCustomerProductListsDetails.data[0],
-            productPromotions: []
+            name: 'product name',
+            price: 25.6,
+            tieredPrices: [
+                {
+                    price: 30.6,
+                    pricebook: 'gbp-m-list-prices',
+                    quantity: 1
+                },
+                {
+                    price: 25.6,
+                    pricebook: 'gbp-m-sale-prices',
+                    quantity: 1
+                },
+                {
+                    price: 15,
+                    pricebook: 'gbp-m-sale-prices',
+                    quantity: 10
+                }
```

Comment:
[NIT] Might be useful to demonstrate in this test case that we take the closest tieredPrice quantity that is lower than the provided quantity:

```
tieredPrices: [
   ...
   {
        price: 10,
        pricebook: 'gbp-m-sale-prices',
        quantity: 15
    }
]

const {listPrice, currentPrice} = getDisplayPrice(data, {quantity: 14})
```

this example shows that even though 14 is closer to 15 than 10, we still choose the tieredPrice of quantity 10

Reply:
test case was added